2024-09-08 14:56:02,628 : INFO : root : Loading config.
2024-09-08 14:56:02,630 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2024-09-08 14:56:02,630 : INFO : ConfigProcessor : Loading config from cache: C:\Users\dorus\AppData\Local\Temp\636ce416879e362e618e72aa23ee7aae.mpf_cache
2024-09-08 14:56:02,687 : WARNING : ConfigProcessor : Config file in cache changed: C:\Users\dorus\OneDrive\Documents\Pinball\PortalPinball-V4.0\machinefolder\modes\attract\config\attract.yaml
2024-09-08 14:56:02,688 : INFO : ConfigProcessor : Loading config from file C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\modes\attract\config\attract.yaml.
2024-09-08 14:56:02,689 : INFO : ConfigProcessor : Loading config: C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\modes\attract\config\attract.yaml
2024-09-08 14:56:02,689 : INFO : ConfigProcessor : Loading config from file C:\Users\dorus\OneDrive\Documents\Pinball\PortalPinball-V4.0\machinefolder\modes\attract\config\attract.yaml.
2024-09-08 14:56:02,690 : INFO : ConfigProcessor : Loading config: C:\Users\dorus\OneDrive\Documents\Pinball\PortalPinball-V4.0\machinefolder\modes\attract\config\attract.yaml
2024-09-08 14:56:02,690 : INFO : ConfigProcessor : Config file cache created: C:\Users\dorus\AppData\Local\Temp\88f25e45fa15743a3d93de98d7c6ca4c.mpf_cache
2024-09-08 14:56:02,690 : INFO : ConfigProcessor : Loading config from cache: C:\Users\dorus\AppData\Local\Temp\d8bc63701c1a2ad1653a8cc5aec9548a.mpf_cache
2024-09-08 14:56:02,690 : INFO : Machine : Mission Pinball Framework Core Engine v0.80.0.dev4
2024-09-08 14:56:02,690 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'platform_integration_test': False, 'jsonlogging': False, 'logfile': 'logs\\2024-09-08-14-56-02-mpf-Gaming_Dorus.log', 'pause': False, 'production': False, 'text_ui': True, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': 'virtual', 'syslog_address': None, 'mc_file_name': None, 'no_sound': False, 'full_logfile_path': 'C:\\Users\\dorus\\OneDrive\\Documents\\Pinball\\PortalPinball-V4.0\\machinefolder\\logs\\2024-09-08-14-56-02-mpf-Gaming_Dorus.log'}
2024-09-08 14:56:02,691 : INFO : Machine : MPF path: C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf
2024-09-08 14:56:02,691 : INFO : Machine : Machine path: C:\Users\dorus\OneDrive\Documents\Pinball\PortalPinball-V4.0\machinefolder
2024-09-08 14:56:02,691 : INFO : Machine : Platform: win32
2024-09-08 14:56:02,691 : INFO : Machine : Python executable location: C:\ProgramData\anaconda3\envs\mpfenv\python.exe
2024-09-08 14:56:02,691 : INFO : Machine : Python version: 3.12.4 (64-bit)
2024-09-08 14:56:02,692 : INFO : Machine : Initializing MPF...
2024-09-08 14:56:02,877 : INFO : EventManager : Event: ======'machine_var_credits_string'====== Args={'value': 'FREE PLAY', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,891 : INFO : EventManager : Event: ======'machine_var_master_volume'====== Args={'value': 0.5, 'prev_value': None, 'change': True}
2024-09-08 14:56:02,891 : INFO : EventManager : Event: ======'machine_var_mpf_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev4', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,891 : INFO : EventManager : Event: ======'machine_var_mpf_extended_version'====== Args={'value': 'Mission Pinball Framework v0.80.0.dev4, Config version:6, Show version: 6, BCP version:1.1', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,891 : INFO : EventManager : Event: ======'machine_var_python_version'====== Args={'value': '3.12.4', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,956 : INFO : EventManager : Event: ======'machine_var_platform'====== Args={'value': 'Windows-11-10.0.22631-SP0', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,956 : INFO : EventManager : Event: ======'machine_var_platform_system'====== Args={'value': 'Windows', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,957 : INFO : EventManager : Event: ======'machine_var_platform_release'====== Args={'value': '11', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,957 : INFO : EventManager : Event: ======'machine_var_platform_version'====== Args={'value': '10.0.22631', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,957 : INFO : EventManager : Event: ======'machine_var_platform_machine'====== Args={'value': 'AMD64', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,957 : INFO : EventManager : Event: ======'machine_var_log_file_path'====== Args={'value': 'C:\\Users\\dorus\\OneDrive\\Documents\\Pinball\\PortalPinball-V4.0\\machinefolder\\logs\\2024-09-08-14-56-02-mpf-Gaming_Dorus.log', 'prev_value': None, 'change': True}
2024-09-08 14:56:02,957 : INFO : EventManager : Event: ======'init_phase_1'====== Args={}
2024-09-08 14:56:02,969 : INFO : EventManager : Event: ======'drop_target_bank_db_dropbank_up'====== Args={}
2024-09-08 14:56:02,969 : INFO : EventManager : Event: ======'init_phase_2'====== Args={}
2024-09-08 14:56:02,970 : INFO : SwitchController : Found active switch: ('2-1-16', <Platform.Virtual>) <switch.s_trough1>
2024-09-08 14:56:02,970 : INFO : SwitchController : Found active switch: ('2-1-17', <Platform.Virtual>) <switch.s_trough2>
2024-09-08 14:56:02,970 : INFO : SwitchController : Found active switch: ('2-1-18', <Platform.Virtual>) <switch.s_trough3>
2024-09-08 14:56:02,970 : INFO : SwitchController : Found active switch: ('2-1-19', <Platform.Virtual>) <switch.s_trough4>
2024-09-08 14:56:02,970 : INFO : EventManager : Event: ======'bcp_connection_attempt'====== Args={'name': 'local_display', 'host': 'localhost', 'port': 5050}
2024-09-08 14:56:02,972 : INFO : BCPClientSocket : Connecting BCP to 'local_display' at localhost:5050...
2024-09-08 14:56:11,775 : ERROR : asyncio : Exception in callback DelayManager._process_delay_callback('draw_screen', <bound method...013C86CD8D50>>)()
handle: <TimerHandle when=151076.135 DelayManager._process_delay_callback('draw_screen', <bound method...013C86CD8D50>>)()>
Traceback (most recent call last):
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\core\delays.py", line 189, in _process_delay_callback
    callback(**kwargs)
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\core\text_ui.py", line 439, in _draw_screen
    self.screen.draw_next_frame()
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\screen.py", line 1690, in draw_next_frame
    event = scene.process_event(event)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\scene.py", line 98, in process_event
    event = effect.process_event(event)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 699, in process_event
    claimed_focus, quit_now = self._handle_desktop_ordering(event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 677, in _handle_desktop_ordering
    effect.blur()
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 540, in blur
    self._layouts[self._focus].blur()
    ~~~~~~~~~~~~~^^^^^^^^^^^^^
IndexError: list index out of range
2024-09-08 14:56:11,792 : INFO : Machine : Starting the main run loop.
2024-09-08 14:56:11,794 : INFO : Machine : Shutting down...
2024-09-08 14:56:11,794 : INFO : EventManager : Event: ======'shutdown'====== Args={}
2024-09-08 14:56:12,817 : ERROR : Machine : Runtime Exception
Traceback (most recent call last):
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\core\machine.py", line 793, in _run_loop
    raise self._exception['exception']
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\core\delays.py", line 189, in _process_delay_callback
    callback(**kwargs)
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\mpf\core\text_ui.py", line 439, in _draw_screen
    self.screen.draw_next_frame()
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\screen.py", line 1690, in draw_next_frame
    event = scene.process_event(event)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\scene.py", line 98, in process_event
    event = effect.process_event(event)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 699, in process_event
    claimed_focus, quit_now = self._handle_desktop_ordering(event)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 677, in _handle_desktop_ordering
    effect.blur()
  File "C:\ProgramData\anaconda3\envs\mpfenv\Lib\site-packages\asciimatics\widgets\frame.py", line 540, in blur
    self._layouts[self._focus].blur()
    ~~~~~~~~~~~~~^^^^^^^^^^^^^
IndexError: list index out of range
2024-09-08 14:56:12,828 : INFO : root : MPF run loop ended.
